debian/patches/0052-add-vf-tonemap-videotoolbox-filter.patch

```diff
@@ -298,26 +300,44 @@ Index: FFmpeg/libavfilter/metal/vf_tonemap_videotoolbox.metal
 +}
 +
 +float3 rgb2yuv(float r, float g, float b) {
-+    if (!is_full_range_out) {
-+        r = r * LIMITED_RANGE + LIMITED_BLACK;
-+        g = g * LIMITED_RANGE + LIMITED_BLACK;
-+        b = b * LIMITED_RANGE + LIMITED_BLACK;
++    float y = (r * yuv_matrix_1[0]) + (g * yuv_matrix_1[1]) + (b * yuv_matrix_1[2]);
++    float u = (r * yuv_matrix_2[0]) + (g * yuv_matrix_2[1]) + (b * yuv_matrix_2[2]);
++    float v = (r * yuv_matrix_3[0]) + (g * yuv_matrix_3[1]) + (b * yuv_matrix_3[2]);
++    if (is_full_range_out) {
++        u += 0.5f;
++        v += 0.5f;
++    } else {
++        if (enable_dither) {
++            y = floor((219.0f * y + 16.0f) + 0.5f) / 255.0f;
++            u = floor((224.0f * u + 128.0f) + 0.5f) / 255.0f;
++            v = floor((224.0f * v + 128.0f) + 0.5f) / 255.0f;
++        } else {
++            y = floor(((219.0f * y + 16.0f) * 256.0f) + 0.5f) / 65535.0f;
++            u = floor(((224.0f * u + 128.0f) * 256.0f) + 0.5f) / 65535.0f;
++            v = floor(((224.0f * v + 128.0f) * 256.0f) + 0.5f) / 65535.0f;
++        }
++    }
++    // in rgb2yuv conversion, enable_dither means output is 8bit
++    if (!enable_dither) {
++        y -= mix(0.0f, 0.0009765774014f, y > 0.0f);
```

Comment:
Will this optimization make a noticeable difference?

Reply:
Actually not. The most problematic problems come from the upstream handling and not this pixel format quantization error.

Reply:
For this particular space probably we don't really need to account for such small offset because the accuracy loss during the computation might be larger than this.

Reply:
So you can consider whether to omit it to make the code a little cleaner.

Reply:
I decided to keep it but with extra documentation.